lms/static/scripts/frontend_apps/components/test/BasicLtiLaunchApp-test.js

```diff
@@ -510,12 +517,127 @@ describe('BasicLTILaunchApp', () => {
         path: '/api/lti/submissions',
-        data: fakeConfig.canvas.speedGrader.submissionParams,
+        data: {
+          submitted_at: undefined,
+          ...fakeConfig.canvas.speedGrader.submissionParams,
+        },
       });
     });
+
+    context(
+      'When configured to wait for first annotation activity before submission',
+      () => {
+        beforeEach(() => {
+          fakeConfig.canvas.speedGrader = {
+            submissionParams: {
+              lis_result_sourcedid: 'modelstudent-assignment1',
+            },
+          };
+          fakeConfig.viaUrl = 'https://via.hypothes.is/123';
+          fakeConfig.hypothesisClient.reportActivity = {
+            method: 'reportActivity',
+            events: ['create', 'edit'],
+          };
+        });
+
+        function getOnActivityCalls(stub) {
+          return stub
+            .getCalls()
+            .filter(call => call.args[0] === 'annotationActivity');
+        }
+
+        it('does not report the submission when the content iframe starts loading', async () => {
+          const wrapper = renderLTILaunchApp();
+          await new Promise(resolve => setTimeout(resolve, 1));
+          assert.isFalse(
+            fakeApiCall.calledWith(
+              sinon.match({ path: '/api/lti/submissions' })
+            )
+          );
+
+          // After the successful API call, the iframe should still be rendered.
+          wrapper.update();
+          assert.isTrue(wrapper.exists('iframe'));
+        });
+
+        it('registers a callback for `annotationActivity`', async () => {
+          renderLTILaunchApp();
+          const annotationActivityCalls = getOnActivityCalls(fakeRpcServer.on);
+          assert.equal(annotationActivityCalls.length, 1);
+          assert.isFunction(annotationActivityCalls[0].args[1]);
+        });
+
+        it('submits a submission when `onAnnotationActivity` callback invoked', async () => {
+          renderLTILaunchApp();
+
+          const annotationActivityCalls = getOnActivityCalls(fakeRpcServer.on);
+          const callback = annotationActivityCalls[0].callback;
+          assert.isFalse(
+            fakeApiCall.calledWith(
+              sinon.match({ path: '/api/lti/submissions' })
+            )
+          );
+          callback('create', {});
+          await new Promise(resolve => setTimeout(resolve, 1));
+
+          assert.calledOnce(fakeApiCall);
+          assert.calledWith(
+            fakeApiCall,
+            sinon.match({ path: '/api/lti/submissions' })
+          );
+
+          const apiCall = fakeApiCall.getCall(0);
+          assert.deepEqual(apiCall.args[0], {
+            authToken: 'dummyAuthToken',
+            path: '/api/lti/submissions',
+            data: {
+              submitted_at: undefined,
+              ...fakeConfig.canvas.speedGrader.submissionParams,
+            },
+          });
+        });
+
+        it('deregisters callback after submission is made', async () => {
```

Comment:
>  I think it would be clearer to test for that directly. 

This is...challenging to do. I have hold of the callback that gets registered, but since `ClientRPC` is mocked I can't `emit` an event (and then testing that a submission does NOT happen) without reworking these tests work.